server/models/Profile.js

```diff
@@ -28,12 +28,13 @@ const profileSchema = new mongoose.Schema({
   },
   birthday: {
     type: Date,
-    default: null
+    default: null,
   },
   photo: {
     type: String,
     default: "",
   },
+  discriminatorKey: "userType",
```

Comment:
I believe this should be passed as an [option](https://mongoosejs.com/docs/guide.html#options) (here it's just another field in the model).